### ISSUE TITLE
Detection of IP address family broken

### ISSUE THREAD

**Issue by vladimiroltean** [troglobit/mtools#8]
Hi Joachim,
mtools used to support auto-detection of IPv4/IPv6 address family by calling inet_pton() in what is now inet_parse(). That no longer works, as of commit 3e7058b86820 ("Massive refactor and code cleanup, preparing for SSM support"), as shown by git bisect.

Good case:
```
root@debian:~/selftests/net/forwarding# msend -g ff0e::181 -I swp0
Now sending to multicast group: ff0e::181
Sending msg 1, TTL 1, to ff0e::181:4444:
```
Bad case:
```
root@debian:~/selftests/net/forwarding# msend -g ff0e::181 -I swp0
Now sending to multicast group: [ff0e::181]:4444
sendto: Address family not supported by protocol
```

I don't understand what was the purpose of sidestepping that address detection mechanism and introducing -4/-6 options?
The kernel selftests use the variant without the -4/-6 switches, and they are currently broken with the latest mtools. Could you please take a look?
Thanks,
Vladimir

**Comment by troglobit** [comment]
Thanks for reporting this, it was an ordering bug!

The -4/-6 is only supposed to kick in for the case `msend -6 -I swp0`, then it selects the IPv6 test group.

**Comment by troglobit** [comment]
Fixed and released as v3.2, available from the usual place, here https://github.com/troglobit/mtools/releases/tag/v3.2

**Comment by vladimiroltean** [comment]
Thanks for fixing it so promptly! I can confirm the kernel selftests pass with tag v3.2.
